app/main.cpp

```diff
@@ -294,23 +295,30 @@ int main( int argc, char *argv[] )
   LocalProjectsManager localProjects( projectDir );
   ProjectModel pm( localProjects );
   Loader loader;
+  AppSettings as;
   LayersModel lm( loader.project() );
   MapThemesModel mtm( loader.project() );
-  AppSettings as;
   std::unique_ptr<MerginApi> ma =  std::unique_ptr<MerginApi>( new MerginApi( localProjects ) );
   MerginProjectModel mpm( localProjects );
   MerginProjectStatusModel mpsm( localProjects );
 
   // Connections
   QObject::connect( &app, &QGuiApplication::applicationStateChanged, &loader, &Loader::appStateChanged );
-  QObject::connect( &loader, &Loader::projectReloaded, &lm, &LayersModel::reloadLayers );
-  QObject::connect( &loader, &Loader::projectReloaded, &mtm, &MapThemesModel::reloadMapThemes );
-  QObject::connect( &mtm, &MapThemesModel::reloadLayers, &lm, &LayersModel::reloadLayers );
   QObject::connect( ma.get(), &MerginApi::syncProjectFinished, &pm, &ProjectModel::addProject );
   QObject::connect( ma.get(), &MerginApi::listProjectsFinished, &mpm, &MerginProjectModel::resetProjects );
   QObject::connect( ma.get(), &MerginApi::syncProjectStatusChanged, &mpm, &MerginProjectModel::syncProjectStatusChanged );
   QObject::connect( ma.get(), &MerginApi::reloadProject, &loader, &Loader::reloadProject );
 
+  QObject::connect( &loader, &Loader::projectReloaded, &mtm, &MapThemesModel::reloadMapThemes );
+  QObject::connect( &mtm, &MapThemesModel::mapThemesReloaded, &as, &AppSettings::reloadDefaultMapTheme );
+  QObject::connect( &as, &AppSettings::reloadDefaultMapThemeSignal, &mtm, &MapThemesModel::updateMapTheme );
```

Comment:
you do not need to name `reloadDefaultMapThemeSignal` with signal suffix

Reply:
reloadDefaultMapThemeSignal and reloadDefaultMapTheme are related (both in AppSettings). Basically the idea is to retrieve name of default map theme and to avoid introducing AppSetting instance to the model therefore the model can be independent.

Same for the retrieving defaultLayer name in layer's model.

The signal suffix or a whole signal name can be replaced though (getDefault* and sendDefault* to keep the relation obvious)

Reply:
if you have getDefault* and sendDefault*, why it cannot be one signal that is connected directly here? If you want to send string in argument list, it does not matter, or does it?